### PR TITLE
fix: don't use binary if not asking for package version

### PR DESCRIPTION
**Pull request by fubuloubu**
fixes: #31

### What I did

fixes: #

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: ape_vyper/compiler.py [hunk 1]
```diff
@@ -102,29 +102,32 @@ def compile(
...
+
+            elif pragma_spec:
+                vyper_version = pragma_spec.select(self.installed_versions)
+
+            elif not self.installed_versions:
+                vyper_version = max(self.available_versions)
+                _install_vyper(vyper_version)
+
+            else:
+                vyper_version = max(self.installed_versions)
```
Comment (fubuloubu) [inline ape_vyper/compiler.py:122]: Just wanted to simplify this logic a bunch

File: ape_vyper/compiler.py [hunk 2]
```diff
@@ -102,29 +102,32 @@ def compile(
...
+                    
+            vyper_binary = vyper_version == self.package_version and shutil.which("vyper") or None
             try:
                 result = vvm.compile_source(
                     source,
                     base_path=base_path,
                     vyper_version=vyper_version,
-                    vyper_binary=shutil.which("vyper") or None,
+                    vyper_binary=vyper_binary,
```
Comment (fubuloubu) [inline ape_vyper/compiler.py:130]: This is the actual fix

File: ape_vyper/compiler.py [hunk 3]
```diff
@@ -102,29 +102,32 @@ def compile(
...
+                vyper_version = pragma_spec.select(self.installed_versions)
+
+            elif not self.installed_versions:
+                vyper_version = max(self.available_versions)
+                _install_vyper(vyper_version)
+
+            else:
+                vyper_version = max(self.installed_versions)
+
+            vyper_binary = vyper_version == self.package_version and shutil.which("vyper") or None
```
Comment (unparalleled-js) [inline ape_vyper/compiler.py:124]: The typing of this variable is a bit confusing. It is either a `False`, `None`,  or a `str` (the path to the binary).

Comment (unparalleled-js) [inline ape_vyper/compiler.py:124]:
```suggestion
            vyper_binary = shutil.which("vyper") or None if vyper_version == self.package_version else None
```

?

Comment (unparalleled-js) [reply ape_vyper/compiler.py:124]:
```python
v = 3 == 1 and "Test"
>>> False
```

**Review by unparalleled-js — COMMENTED** [review]
I think we should always pass in `None` or a `str` for the `vyper_binary` argument.
https://github.com/vyperlang/vvm/blob/master/vvm/main.py#L30

File: ape_vyper/compiler.py (hunk 3 above)
Comment (unparalleled-js) [reply ape_vyper/compiler.py]: Oh oops! I see

**Pull request merged**